doc/theming/htmx.rst

```diff
@@ -0,0 +1,156 @@
+In UI terms, the **Follow** / **Unfollow** logic is just a div containing a button that
+allows the user to follow/unfollow a dataset plus a counter that shows the number of
+followers. The div is displayed in the dataset page.
+
+This is a small interactive action and we do not want a typical full refresh of the page. It
+doesn't make any sense to reload the whole page just to update the number of followers and the
+button. This is a perfect use case for ``htmx``.
+
+What we need to achieve this behaviour is:
+  1. A HTML structure that encapsulates the follow/unfollow UI in a single HTML element (so it can be replaced).
+  2. A way to trigger a call to the endpoint when the user clicks on the button and replace the element with the new content.
+  3. A new endpoint that covers the backed logic and returns just enough HTML to replace the HTML element.
+
+
+1. HTML structure
+
+The HTML structure is very simple. We just need a div that contains the button and the counter. In our case, and
+to respect the current CKAN UX, we have the ``package/snippets/info.html`` snippet that is used to render this
+information. We just need to make sure that the HTML element that we want to replace has an id. In our case,
+we just add and id to it: ``package-info``.
+
+.. code-block:: jinja
+
+    <!-- package/snippets/info.html -->
+    {% block package_info %}
+      {% if pkg %}
+        <section id="package-info" class="module module-narrow">
+        <!-- Rest of the snippet -->
+        </section>
+      {% endif %}
+    {% endblock %}
+
+
+2. Triggering a call to the endpoint
+
+We need to trigger a call to the endpoint when the user clicks on the button. We can do this by adding the
+``hx-post`` attribute to the button. The ``hx-post`` attribute defines the URL that will be called when the
+user clicks on the button. In our case, we want to call the ``/dataset/follow/<dataset-id>`` endpoint, so
+we can use the ``h.url_for`` helper to generate the URL.
+
+
+.. code-block:: jinja
+
+    <a class="btn btn-danger" hx-post="{{ h.url_for('dataset.follow', id=pkg.id) }}" hx-target="#package-info">
+      <i class="fa-solid fa-circle-plus"></i>
+      Follow
+    </a>
+
+In addition to the ``hx-post`` attribute, we also need to define the ``hx-target`` attribute. The ``hx-target``
+attribute defines the HTML element that will be replaced with the HTML returned by the endpoint. In our case,
+we want to replace the ``package-info`` element, so we can use the ``#package-info`` selector.
+
+3. The endpoint
+
+The last step is to implement the endpoint that will be called when the user clicks on the button. In our case,
+we want to call the ``/dataset/follow/<dataset-id>`` endpoint. This endpoint is already implemented in CKAN.
+We just need to make sure that, under this new context, it should return only the HTML that we want to display in the page
+instead of rendering again the whole dataset page.  We achieve that by making it sure that we return the snippet that
```

Comment:
```suggestion
We need to make sure that, under this new context, it should return only the partial HTML that we want to insert into the page
instead of rendering the whole dataset page again.  We achieve that by making it sure that we return the snippet that
```